ordered_product/tests.py

```diff
@@ -0,0 +1,127 @@
+                            location="Kiryat Shmona",
+                            date=datetime.date(2022, 12, 30))
+
+
+@pytest.fixture
+def ordered_product0():
+    sup = Supplier(user_name="WorstSup",
+                   first_name="Yuval",
+                   last_name="Cohen",
+                   password="1234",
+                   user_type="supplier",
+                   business_name="bestSup")
+    sup.save()
+    client = Client(user_name="meitar1996",
+                    first_name="Meitar",
+                    last_name="Rizner",
+                    password="123456",
+                    user_type="client",
+                    area="North Israel")
+    client.save()
+    product = Product(qr_code="Q5o76MbdiNbXprNEnHfpcGWFp1CMF8XY",
+                      product_name="Apple",
+                      description="Sweety!")
+    product.save()
+    sup_product = SupplierProduct(suppler_product_id=1,
+                                  qr_code=product,
+                                  user_name=sup,
+                                  price=10,
+                                  quantity=50)
+    sup_product.save()
+    del_location = DeliveryLocation(user_name=sup, location="Kiryat Shmona", date=datetime.date(2022, 12, 30))
+    del_location.save()
+    return OrderdProduct(
+        ordered_product_id=1,
+        delivery_location_id=del_location,
+        user_name=client,
+        supplier_product_id=sup_product,
+        quantity=3)
+
+
+@pytest.fixture
+def saved_ordered_product0(ordered_product0):
+    ordered_product0.save_ordered_product()
+    return ordered_product0
+
+
+class TestOrderedProductModel():
+    @pytest.mark.django_db()
+    def test_save_and_delete_ordered_product(self, saved_ordered_product0):
+        temp = OrderdProduct(ordered_product_id=saved_ordered_product0.ordered_product_id,
+                             delivery_location_id=saved_ordered_product0.delivery_location_id,
+                             user_name=saved_ordered_product0.user_name,
+                             supplier_product_id=saved_ordered_product0.supplier_product_id,
+                             quantity=saved_ordered_product0.quantity)
+        assert saved_ordered_product0 in OrderdProduct.objects.all()
+        saved_ordered_product0.delete_ordered_product()
+        assert saved_ordered_product0 not in OrderdProduct.objects.all()
+        temp.save()
+        assert temp in OrderdProduct.objects.all()
+
+    @pytest.mark.django_db()
+    def test_filter_by_ordered_prodct_ID(self, saved_ordered_product0):
+        assert saved_ordered_product0 not in OrderdProduct.filter_ordered_prodct_ID(1000)
+        filt = OrderdProduct.filter_ordered_prodct_ID(saved_ordered_product0.ordered_product_id)
+        assert [saved_ordered_product0] == list(filt)
+
+    @pytest.mark.django_db()
+    def test_filter_by_user_name(self, saved_ordered_product0):
+        assert saved_ordered_product0 not in OrderdProduct.filter_user_name("")
```

Comment:
Why do we need to `list()` the result at the positive test but not on the negative test?